### PR TITLE
Fix the abnormal damage of shotgun at point blank

### PR DESCRIPTION
**Pull request by wwkk222208**
#### Summary
None

#### Purpose of change
Shot_bird at point blank can cause damage to high armor targets, but damage of 00_short is lower than expected. The purpose of this PR is to fix this problem.

On the one hand，it seems that most people don’t think a shotgun is less lethal at point blank (https://www.quora.com/Is-a-shotgun-less-lethal-from-the-point-blank-range)(https://www.quora.com/What-happens-when-a-person-is-shot-at-point-blank-range). According to their analysis, this is likely to be true. In #53142, I increased shot spread of 00_shot and shot_bird for simulation, which caused the shotgun’s performance at a distance to decrease. If the shotgun damage is not enough to kill unarmored zombies in one shot at range blank, then their practicality will become relatively low. In terms of simulation and gameplay, it seems necessary to ensure the lethality of shotguns at close range.

On the other hand, through my tests, I found that even if the cloud of projectiles are not considered as a complete bullet, all of them can still hit the target.

In a nutshell, my suggestion is:
1.shot_birds should not be lethal for armored target at point blank
2.00_shot should be lethal for unarmored target at point blank

#### Describe the solution
Delete the setting of taking shotgun as bullet when the distance is 1.

#### Describe alternatives you've considered
We can also consider adjusting the damage of shot in json without changing the C++ code, if It is necessary to treat shot differently at point blank.

#### Testing
A npc (all stats = 10 level of all skills=5) shoot different monsters at different range with shotgun_s without recoil 10000 times. Caculate average damge.
The results show that damage at point blank and damage at close range (range=5) are almost the same, which implies that all projectiles hit the target.

### CODE REVIEW

**Comment by catdach** [comment]
I'm not fully understanding how these numbers translate into damage so forgive me if I'm just misunderstanding but, why is birdshot's point blank damage being reduced so much on enemies with trivial and low armor values? I know it's weaker than buckshot, but seems strange that a number is dropping by 19 for an enemy with 1 bullet armor value.

**Comment by wwkk222208** [comment]
> I know it's weaker than buckshot, but seems strange that a number is dropping by 19 for an enemy with 1 bullet armor value.
This is because shot_bird actually fires 100 projectiles that can only cause 0.25 damage.

**Comment by catdach** [comment]
> > I know it's weaker than buckshot, but seems strange that a number is dropping by 19 for an enemy with 1 bullet armor value.
> 
> This is because shot_bird actually fires 100 projectiles that can only cause 0.25 damage, while 00_shot fires 9 projectiles that can cause 15 damage each.

I thought shot type ammo was intentionally made to act as a single damage instance at point blank as of #52583?

**Comment by wwkk222208** [comment]
I already know Kevin's intentions from the code and comments.
But he also doesn't want shot_bird to cause damage to armored units (you can find this in the comments).
On the one hand，it seems that most people don’t think a shotgun is less lethal at point blank（https://www.quora.com/Is-a-shotgun-less-lethal-from-the-point-blank-range. According to their analysis, this is likely to be true
On the other hand, through my tests, I found that even if the cloud of projectiles are not considered as a complete bullet, all of them can still all hit the target.
I also propose another solution, which is to keep the current code and adjust the damage when the shot in json is regarded as a whole.

**Review by kevingranade — CHANGES_REQUESTED** [review]
I don't understand what you're getting at here, this behavior is 100% intentional and based on the most rigorous source I can find.  on top of that, I'm not able to follow your argument for why you want to make this change, you seem to be vacillating back and forth between "shot should be lethal at point blank" and "shot should not be lethal at point blank".

**Comment by wwkk222208** [comment]
Sorry, my expression is not clear enough.
My suggestion is:
1.shot_birds should not be lethal for armored target at point blank
2.00_shot should be lethal for unarmored target at point blank

**Comment by wwkk222208** [comment]
I agree with "the pellets remain tightly clustered with the entire load functioning as a unit", but I am not sure whether it is reasonable to consider the pellets as a whole when determining damage

**Comment by wwkk222208** [comment]
My original intention is that the game is more realistic and interesting. If you think my opinion is inappropriate, I am glad to accept it.

**Comment by kevingranade** [comment]
>  1.shot_birds should not be lethal for armored target at point blank

The source says it is though, it's just that simple, it clearly states that the mass of pellets acts as a single bullet at that range.

If you think about it, a lead slug deforms extremely readily, so there is not as much difference between a cloud of pellets that happen to be very close to each other and a solid lump of lead as it might seem, both are going to deform and flow as they impact the target, but also as penetration occurs, the leading pellets or front of the slug is making a channel that the remainder will prefer to follow.

**Comment by wwkk222208** [comment]
Sounds great, you convinced me
